Second batch of CTS test refactorings

clumens opened the pull request:
This is just like the previous PR, but for the next set of test cases.  All have been run and work (SplitBrain has a lot of BadNews, and RollingUpgrade is disabled so hasn't actually been tested).

kgaillot commented on python/pacemaker/_cts/tests/resourcerecover.py:
```diff
@@ -39,35 +46,33 @@ def __call__(self, node):
+            self.debug("Failing %s" % rsc.id)
         else:
-            self.debug("Failing " + rsc.id + " (also known as " + rsc.clone_id + ")")
+            self.debug("Failing %s (also know as %s)" % (rsc.id, rsc.clone_id))
```
known

kgaillot commented on python/pacemaker/_cts/tests/splitbraintest.py:
```diff
@@ -13,16 +12,30 @@
-       check the two nodes both take over the resource'''
-    def __init__(self,cm):
-        CTSTest.__init__(self,cm)
+    """ A concrete test that create a split brain cluster and verifies that
```
creates

kgaillot commented on python/pacemaker/_cts/tests/quorumfailure.py:
```diff
@@ -0,0 +1,56 @@
+
+
+class QuorumFailure(CTSTest):
+    '''Set up a custom test to cause quorum failure issues for Andrew'''
```
Despite this comment, a better name for this test would be something like ResyncCIB. It wipes the CIB from one node, starts it (without quorum), then ensures it picks it up correctly from the other nodes when they join.

clumens replied:
Fixed.

kgaillot commented on python/pacemaker/_cts/tests/rollingupgradetest.py:
```diff
@@ -0,0 +1,103 @@
+from pacemaker._cts.tests.stoptest import StopTest
+
+
+class RollingUpgradeTest(CTSTest):
```
Let's just kill this test. I don't see any way to get rpm-dir, current-version, or previous-version into _env, and the lab is not guaranteed to work across versions since it relies on specific log patterns that sometimes change release to release. Not to mention it assumes RPM packaging. I think manually testing rolling upgrades is probably the best approach.

clumens replied:
Cut.

kgaillot reviewed (commented):
Whew

clumens commented:
Added a patch to the front to fix a bug in the RemoteStonithd test introduced in the last PR, and another patch to remove RollingUpgrade.  And then I removed the other RollingUpgrade patches.

clumens commented:
> Whew

We're almost there!  After this, there's just one more small PR and then CTStests.py is done.  Then I think the PRs should be getting smaller from there.  The tests were the most complicated.
